src/ima-parser.js

```diff
@@ -7,6 +9,41 @@ export default class IMAParserPlugin extends CorePlugin {
 
   constructor(core) {
     super(core)
-    console.log(`${this.name} plugin loaded!`)
+    this._VMAPHandler = new VMAPManager()
+    this._VASTHandler = new VASTManager()
+  }
+
+  /**
+   * Get a list of AdBreaks from the received URL.
+   * @param {String} url Ad Server URL to fetch VMAP XML.
+   * @returns {Promise} Promise resolved with one array of AdBreaks or one error.
+   */
+  requestAdBreaks(url) {
+    return this._VMAPHandler.request(url)
+      .then(xmlPlainText => this._VMAPHandler.process(xmlPlainText))
+      .then(rawAdBreak => this._VMAPHandler.filterRawData(rawAdBreak))
+      .then(adBreaks => {
+        Log.info(this.name, 'Available adBreaks: ', adBreaks)
+        return adBreaks
+      })
+      .catch(error => {
+        Log.error(this.name, 'Fail to request VMAP: ', error)
+        return Promise.reject(error)
+      })
+  }
+
+  /**
+   * Get a list of ads available from the received AdBreak.
+   * @param {Object} adBreak An adBreak Instance.
+   * @returns {Promise} Promise resolved with one array of current available ads of the respected AdBreak.
+   */
+  requestAds(adBreak) {
```

Comment:
I liked that you separated the request for ads from the request for ad breaks. This way we can lazy load the ads as necessary :D